ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Selection-Tool/Lasso-Selection-Tool/lasso-tool.spec.ts

```diff
@@ -0,0 +1,258 @@
+  delay,
+  DELAY_IN_SECONDS,
+  waitForIndigoToLoad,
+} from '@utils';
+import { getAtomByIndex } from '@utils/canvas/atoms';
+import { getBondByIndex } from '@utils/canvas/bonds';
+
+test.describe('Lasso Selection tool', () => {
+  test.beforeEach(async ({ page }) => {
+    await page.goto('');
+    await waitForIndigoToLoad(page);
+  });
+
+  test.afterEach(async ({ page }) => {
+    await takeEditorScreenshot(page);
+  });
+
+  const xDelta = 30;
+  const yDelta = 60;
+  const modifier = getControlModifier();
+
+  const selectLasso = async (page: Page) => {
+    await page.getByTestId('select-rectangle').click();
+    await page.getByTestId('select-lasso').click();
+  };
+
+  async function selectObjects(page: Page, xAxis: number, yAxis: number) {
+    const point = await getCoordinatesOfTheMiddleOfTheScreen(page);
+    await page.mouse.move(point.x - xAxis, point.y - yAxis);
+    await page.mouse.down();
+    await page.mouse.move(point.x + xAxis, point.y - yAxis);
+    await page.mouse.move(point.x + xAxis, point.y + yAxis);
+    await page.mouse.move(point.x - xAxis, point.y + yAxis);
+    await page.mouse.up();
+    return point;
+  }
+
+  async function clickCanvas(page: Page) {
+    await page.mouse.click(300, 200);
+  }
+
+  test('Selection of atom/bond/molecule', async ({ page }) => {
+    /*
+     * Test case: EPMLSOPKET-1338
+     * Description: Hover and selection of atom/bond/molecule
+     */
+    await openFileAndAddToCanvas('KET/two-benzene-with-atoms.ket', page);
+    const atomPoint = await getAtomByIndex(page, { label: 'C' }, 0);
+    await page.mouse.move(atomPoint.x, atomPoint.y);
+    await page.mouse.click(atomPoint.x, atomPoint.y);
+    await clickCanvas(page);
+
+    const bondPoint = await getBondByIndex(page, {}, 0);
+    await page.mouse.move(bondPoint.x, bondPoint.y);
+    await page.mouse.click(bondPoint.x, bondPoint.y);
+    await clickCanvas(page);
+
+    await page.keyboard.down('Shift');
+    await page.mouse.click(atomPoint.x, atomPoint.y);
+    await page.mouse.click(bondPoint.x, bondPoint.y);
+    await page.keyboard.up('Shift');
+    await clickCanvas(page);
+
+    await selectLasso(page);
+    await selectObjects(page, 300, 200);
+  });
+
+  test('Drag atom/bond/molecule', async ({ page }) => {
+    /**
+     * Test case: EPMLSOPKET-1339
+     * Description: Atom/bond/molecule is moved to another place. Structure is not changed. Only selected part changed their place.
+     */
+    await openFileAndAddToCanvas('KET/two-benzene-with-atoms.ket', page);
+    await selectLasso(page);
+    const point = await selectObjects(page, 100, 100);
+    await clickOnAtom(page, 'C', 5);
+    await dragMouseTo(point.x + xDelta, point.y - yDelta, page);
+  });
+
+  test('Select the reaction components', async ({ page }) => {
+    /**
+     * Test case: EPMLSOPKET-1340
+     * Description: Hover and selection of plus sign and reaction arrow
+     */
+    await openFileAndAddToCanvas('Rxn-V2000/reaction_4.rxn', page);
```

Comment:
i propose to use more meaningful names for files. Do we need additional files? May be we can use existing reaction files?